File: mobile_client/app/src/main/java/wpam/mobile_client/protocol/BlindsUPOnTimeCommand.java
package wpam.mobile_client.protocol;

import java.util.ArrayList;

public final class BlindsUPOnTimeCommand extends Command
	{
		private int epochTime_;

		public BlindsUPOnTimeCommand(int epochTime)
		{
			super(10, CommandType.BLINDS_UP_ON_TIME);
			this.epochTime_ = epochTime;
		}
		public final void close()
		{
			super.close();
		}

		@Override
		public ArrayList<Integer> getFrameBytes()
		{
			initializeDataSize();
			ArrayList<Integer> frame = getHeader();
			frame.add(commandType_.getValue());
			BytesConverter.appendUINT32toVectorOfUINT8(epochTime_, frame);

			return frame;
		}

		@Override
		public String getName()
		{
			return new String("BlindsUPOnTimeCommand");
		}
		@Override
		public void accept(CommandVisitor visitor)
		{
			visitor.visit(this);
		}

		public void setEpochDateAndTime(int dateAndTime)
		{
            epochTime_ = dateAndTime;
		}

		public int getEpochDateAndTime()
		{
			return this.epochTime_;
		}

		@Override
		protected void initializeDataSize()
		{
			setDataSize(5);
		}
	}

File: server/src/protocol/BlindsUPOnTimeCommand.cpp
#include "BlindsUPOnTimeCommand.h"
#include "CommandVisitor.h"

#include "BytesConverter.h"

using namespace std;
using namespace utility;
using namespace communication;

BlindsUPOnTimeCommand::BlindsUPOnTimeCommand(uint32_t epochTime)
        : Command(10, CommandType::BLINDS_UP_ON_TIME),
          epochTime_(epochTime)
{}

BlindsUPOnTimeCommand::~BlindsUPOnTimeCommand()
{}

vector<uint8_t> BlindsUPOnTimeCommand::getFrameBytes()
{
    initializeDataSize();

    vector<uint8_t > frame = getHeader();
    frame.push_back(static_cast<uint8_t>(commandType_));

    BytesConverter::appendUINT32toVectorOfUINT8(epochTime_, frame);

    return frame;
}

string BlindsUPOnTimeCommand::getName()
{
    return string("BlindsUPOnTimeCommand");
}

void BlindsUPOnTimeCommand::accept(CommandVisitor &visitor)
{
    visitor.visit(*this);
}

void BlindsUPOnTimeCommand::initializeDataSize()
{
    uint16_t dataSize = sizeof(commandType_);
    dataSize += sizeof(epochTime_);

    setDataSize(dataSize);
}

void BlindsUPOnTimeCommand::setEpochDateAndTime(uint32_t dateAndTime)
{
    epochTime_ = dateAndTime;
}

uint32_t BlindsUPOnTimeCommand::getEpochDateAndTime() const
{
    return epochTime_;
}
